Libraries/Opc.Ua.Server/Subscription/MonitoredItem.cs

```diff
@@ -1225,8 +1253,9 @@ public virtual bool Publish(OperationContext context, Queue<EventFieldList> noti
                     return false;
                 }
 
-                // only publish if reporting.
-                if (!IsReadyToPublish)
+                // only publish if reporting or resending data
+                ResendDataState selectedToResendData = (ResendDataState)Interlocked.CompareExchange(ref m_resendDataState, (int)ResendDataState.NonResendData, (int)ResendDataState.ResendData);
```

Comment:
technically this interlock executes under a lock and the getter/setter is unprotected so the interlocked is not necessary here, unless the setter also uses interlocked.